### PR TITLE
Acceptance Stats Tool

### PR DESCRIPTION
**Pull request by ArsalaanAli**
closes (Github Issue number)

# Quick Overview of Changes

- tool for CDs to tune acceptance stats
- 
image

# Checklist

- [ ] If any frontend changes were made, include a screenshot/demo in the overview
- [ ] Checked all uses of changed component(s)/function(s)
- [ ] If any changes were made to our backend services, at least one (happy path) unit test was added OR a `TODO` comment was added to create one.
- [ ] Check that CI is passing.
- [ ] If no (re-)reviews after 1-2 days, ping the web leads on the discord to remind them to review the PR.

@basokant @ArsalaanAli

### CODE REVIEW

**Comment by hunterchen7** [comment]
this is nice

**Review by basokant — DISMISSED** [review]
lgtm lets merge this in

File: src/pages/internal/acceptance-stats.tsx [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        a.avgPassionRating * passionWeight;
+      const bTotal =
+        b.avgOriginalityRating * originalityWeight +
+        b.avgTechnicalityRating * technicalityWeight +
+        b.avgPassionRating * passionWeight;
+      return bTotal - aTotal;
+    });
+
+    setSortedUserData(sortedData);
+  };
```
Comment (basokant) [inline src/pages/internal/acceptance-stats.tsx:62]: cant we just do
```suggestion
 const sortedData = useMemo(() => {
    return [...allUserData].sort((a, b) => {
      if (a.referral && !b.referral) return -1; // a comes first
      if (!a.referral && b.referral) return 1; // b comes first

      // Apply the weightings
      const aTotal =
        a.avgOriginalityRating * originalityWeight +
        a.avgTechnicalityRating * technicalityWeight +
        a.avgPassionRating * passionWeight;
      const bTotal =
        b.avgOriginalityRating * originalityWeight +
        b.avgTechnicalityRating * technicalityWeight +
        b.avgPassionRating * passionWeight;
      return bTotal - aTotal;
     });
   }, [allUserData])
```

Comment (basokant) [reply src/pages/internal/acceptance-stats.tsx:62]: also remove the useEffect

File: src/pages/internal/acceptance-stats.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    undefined,
+    {
+      refetchOnWindowFocus: false, // Disable refetching on window focus
+      refetchOnMount: false, // Disable refetching on mount
+    },
+  );
+
+  useEffect(() => {
+    applyWeightingsAndSort();
+  }, [allUserData]);
```
Comment (basokant) [inline src/pages/internal/acceptance-stats.tsx:31]:
```suggestion
```

**Review by basokant — APPROVED** [review]

**Pull request closed without merging**